packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1006,11 +1006,33 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
   }
 
   void _resolveImage() {
-    final ImageStream newStream =
-      widget.image.resolve(createLocalImageConfiguration(
+    final DeferringImageProvider<dynamic> provider = DeferringImageProvider<dynamic>(
+      imageProvider: widget.image,
+      getNextAction: () {
+        // Did this element get disposed of (scrolled too far out of view before
+        // scrolling slowed down enough to catch up)?
+        if (!mounted) {
+          return DeferringImageProviderAction.cancel;
+        }
+        // Is it scrolling so fast that the image is unlikely stay around long
+        // enough to be worth decoding? If so, wait until the next task loop
+        // to see if we've slowed down or disappeared.
+        // It's safe to use these calls as point-in-time - do not subscribe to
+        // updates of MediaQuery or Scrollable.
+        final double maxPhysicalDimension = WidgetsBinding.instance.window.physicalSize.longestSide;
```

Comment:
Nit: being explicit on units here can help verify the math.

Reply:
Are you asking for a different variable name or something else?

Reply:
Yep, putting the units in the variable name is the easiest change.

Reply:
Ahh sure.  That makes sense.

Reply:
Just to be a bit more obvious.  I've seen velocity calculations get messed up when people mix logical pixels and hardware pixels.

Reply:
I am intentionally doing that but I'll add some explanation.  

Basically, we want something a factor bigger than logical pixels, and I'm assuming this is safe because higher dpi should mean more memory available on device.

Reply:
Pushed up some changes to the docs and this property name.